tests/unit/test_application.py

```diff
@@ -0,0 +1,88 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for application classes."""
+
+import pytest
+from craft_providers import bases
+
+from snapcraft import application
+from snapcraft.models.project import Architecture
+
+
+@pytest.fixture(
+    params=[
+        ["amd64"],
+        ["riscv64"],
+        ["amd64", "riscv64", "s390x"],
+        [Architecture(build_on="amd64", build_for="riscv64")],
+    ]
+)
+def architectures(request):
```

Comment:
Should we test the `all` keyword for architectures as well?

Reply:
I'm not sure what you mean here. These are the architectures that go into a project, so there wouldn't be an "all".

Reply:
Snapcraft accepts `build-for: all` in a project, so my question is if we should test `[Architecture(build_on="amd64", build_for="all")],` as part of this PR.

Reply:
Ahh, that makes sense, thanks. I've added a test at the bottom of this file for that.